### PR TITLE
 set up associations

### PR DESCRIPTION
**Pull request by mrparkersson**
Pull request
Amo Parker Prince

What does this PR do?

This is a ruby work to Set up associations between classes and objects.

Description of the task to be completed

Implement the classes to represent students and teachers.

Used Ruby in creating all components

created test.MD files that have been ignored
How should it be tested
By running npm test in the terminal.

clone the repository
git clone https://github.com/mrparkersson/school-library
navigate to the folder
cd reactcapstone
checkout the feature

### CODE REVIEW

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Parker, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] in the `Student` class there should be a method that serves as the 'belongs-to' side (a student belongs to a classroom). please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) with examples.
- [ ] in the `Person` class Getters and Setters for @rentals should be present.
- [ ] in the `Person` class @rentals should be initialized, have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for examples.
- [ ] in the `Person` class A method for adding rentals should be present. please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for examples.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Book
+  def initialize(title, author)
+    @title = title
+    @author = author
+  end
+
+  attr_accessor :title, :author
```
[the remark this answers is not in the document]
Comment (Dipeshtwis) [reply book.rb:10]: - [ ] Here make sure Getters and Setters for `@rentals` should be present. You have set only Getters.

Hint on how to set getter and setter.
`attr_accessor :rentals`

File: person.rb [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+require './rentals'
+
+class Person < Nameable
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..9999)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    super()
+  end
```
Comment (Dipeshtwis) [inline person.rb:11]: - [ ] In the Person class `@rentals` should be initialized, 

```ruby
def initialize(age, name = "Unknown", parent_permission = true)
    .
    .
    # *@rentals should be initialized. Link to the example about constructors in a previous lesson https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars
    @rentals = []
  end
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @mrparkersson, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**_Check the comments under the review_**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+
+  def correct_name
+    @name
   end
 
-  attr_reader :id
+  attr_reader :id, :rentals
   attr_accessor :name, :age
```
[the remark this answers is not in the document]
Comment (Kingstalux) [reply person.rb]: resolved 👍

File: person.rb (hunk 2 above)
Comment (Kingstalux) [reply person.rb]: resolved 👍

File: classroom.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  attr_accessor :label
+  attr_reader :students
```
[the remark this answers is not in the document]
Comment (Kingstalux) [reply classroom.rb]: resolved 👍

**Review by Kingstalux — APPROVED** [review]
Hi @mrparkersson,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**